### PR TITLE
Suppress `lint:js` warnings in CI and GitHub PRs

### PR DESCRIPTION
**Pull request by tomalec**
Add returned type annotations to `packages/admin-e2e-tests/src/elements/*.ts`,
to suppress CI and GitHub warnings.

### Accessibility

-   [n/a] I've tested using only a keyboard (no mouse)
-   [n/a] I've tested using a screen reader
-   [n/a] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
-   [n/a] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Before 
A list of warnings in GH PR
```
✖ 431 problems (0 errors, 431 warnings)
```

#### After
image

https://github.com/woocommerce/woocommerce-admin/runs/4461433270?check_suite_focus=true#step:5:742
```
✖ 329 problems (0 errors, 329 warnings)
```

### Detailed test instructions:

1. Open the ["Files changed" tab](https://github.com/woocommerce/woocommerce-admin/pull/8020/files)
2. See fewer warnings.
3. Open [GitHub action run log](https://github.com/woocommerce/woocommerce-admin/actions/runs/1555284320),
4. See fewer lint warnings.
5. `npm run lint:js`
6. See fewer lint warnings.

### CODE REVIEW

File: packages/admin-e2e-tests/src/utils/actions.ts
Comment (ilyasfoo) [inline packages/admin-e2e-tests/src/utils/actions.ts:106]: Same as above

```suggestion
): Promise< unknown > => {
```

File: packages/admin-e2e-tests/src/utils/actions.ts
Comment (ilyasfoo) [inline packages/admin-e2e-tests/src/utils/actions.ts:91]: I think this can be simplified to just `unknown` since it's a [top type](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#example-3)
```suggestion
const getInputValue = async ( selector: string ): Promise< unknown > => {
```

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Sorry for waiting on this for so long, this looks great. Only a nitpick on `unknown` type union

**Comment by adrianduffell** [comment]
Hi @tomalec, just checking whether you've seen the feedback from @ilyasfoo ?

We're planning to [merge Woo Admin to Core](https://developer.woocommerce.com/2021/10/14/rfc-woocommerce-admin-is-merging-into-core/) soon, and ideally it would be best have this PR completed before then.

**Comment by tomalec** [comment]
@ilyasfoo thanks for the suggestion, committed.
@adrianduffell  thanks for the ping.

Sorry, you both for the delay but I was AFK for pretty long.

I merged the `main` branch. There are a few new functions added without annotations, I'd leave them as is to limit the number of not reviewed changes and speed up the PR merge.

**Review by chihsuan — APPROVED** [review]
@tomalec Thanks for fixing the lint warnings. LGTM. 👍

**Comment by tomalec** [comment]
Thanks you @chihsuan for the review, but it seems I still need approval from @ilyasfoo  before merging 
image

**Comment by adrianduffell** [comment]
@tomalec You should be able to merge this now. I've dismissed @ilyasfoo’s requested changes.

**Pull request merged**
